cognite/client/data_classes/_base.py

```diff
@@ -130,14 +131,35 @@ def dump(self, camel_case: bool = False) -> dict[str, Any]:
         return basic_instance_dump(self, camel_case=camel_case)
 
     @classmethod
-    def load(
-        cls: type[T_CogniteResource], resource: dict | str, cognite_client: CogniteClient | None = None
-    ) -> T_CogniteResource:
-        if isinstance(resource, dict):
-            return fast_dict_load(cls, resource, cognite_client=cognite_client)
-        elif isinstance(resource, str):
-            return cls.load(json.loads(resource), cognite_client=cognite_client)
-        raise TypeError(f"Resource must be json str or dict, not {type(resource)}")
+    @final
+    def load(cls: type[Self], resource: dict | str, cognite_client: CogniteClient | None = None) -> Self:
```

Comment:
I think this one is redundant
```suggestion
    def load(cls, resource: dict | str, cognite_client: CogniteClient | None = None) -> Self:
```